drivers/input/keyboard_upper.c

```diff
@@ -123,7 +123,8 @@ static int keyboard_open(FAR struct file *filep)
 
   /* Initializes the buffer for each open file */
 
-  ret = circbuf_init(&opriv->circ, NULL, upper->nums);
+  ret = circbuf_init(&opriv->circ, NULL,
+                     upper->nums * sizeof(struct keyboard_event_s));
```

Comment:
circbuf still needs to be released in failure case

Reply:
@HongChao6 please create a new patch fix the leak at line 142.